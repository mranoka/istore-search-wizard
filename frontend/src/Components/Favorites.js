import React from 'react';
import Button from 'react-bootstrap/Button';
import Container from 'react-bootstrap/Container';
import './Styles.css';

export default class Favorites extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            list: []
        }
        this.handleDelete = this.handleDelete.bind(this);
    }
    // handles deletion of selected item
    handleDelete(e) {
        fetch('/delete', {
            method: "DELETE",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify({
                itemName: e.target.value
            })
        }).then(() => '', err => console.log(err));
        alert(`${e.target.value} has been DELETED!`)
        window.location.reload();
    }
    // retrieves data and music that user has added to favorites

    // retrieves data each time component mounts
    componentDidMount() {
        fetch('/get').then(res => res.json())
            .then(response => {
                console.log(response)
                this.setState({
                    list: response
                })
            }, error => console.log(error));
    }

    render() {
        // compile items in database into a list
        const favItems = this.state.list.map((item, index) => {
            return (
                <>
                    {item.albumArt &&
                        <>
                            <Container className='fav-container'>
                                <div>
                                    <img src={item.albumArt} alt={item.songName} />
                                </div>
                                <ul className='fav-text'>
                                    <li key={index + 10} className='data-list'>{item.artistName}</li>
                                    <li key={index + 12} className='data-list'>{item.songName}</li>
                                    <li key={index + 14} className='data-list'>{item.releaseDate}</li>
                                </ul>
                                <Button onClick={this.handleDelete} value={item.songName} variant="danger">Delete</Button>
                                <hr />
                            </Container>
                        </>
                    }

                    {item.paperBack &&

                        <>
                            <Container className='fav-container'>
                                <div>
                                    <img src={item.paperBack} alt={item.title} />
                                </div>
                                <ul className='fav-text'>
                                    <li key={index + 1} className='data-list'>{item.author}</li>
                                    <li key={index + 2} className='data-list'>{item.title}</li>
                                    <li key={index + 4} className='data-list'>{item.releaseDate}</li>
                                </ul>
                                <Button onClick={this.handleDelete} value={item.title} variant="danger">Delete</Button>
                                <hr />
                            </Container>
                        </>}


                </>
            )

        })
        return (
            <div>
                {favItems}
            </div>
        );
    }

}