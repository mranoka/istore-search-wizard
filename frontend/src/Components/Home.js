import React from 'react';
import 'isomorphic-fetch';
import Card from 'react-bootstrap/Card';
import Container from 'react-bootstrap/Container';
import Button from 'react-bootstrap/Button';
import Music from './Music';
import Books from './Books';
import Favorites from './Favorites';
import './Styles.css';

export default class Home extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            renderSentinel: '',
            searchSentinel: '',
            music: '',
            books: '',
            media: '',
            search: '',
            results: '',
            favList: []
        }
        this.handleInput = this.handleInput.bind(this);
        this.handleMusic = this.handleMusic.bind(this);
        this.handleBooks = this.handleBooks.bind(this);
        this.handleSearch = this.handleSearch.bind(this);
        this.handleMusicAdd = this.handleMusicAdd.bind(this);
        this.handleBooksAdd = this.handleBooksAdd.bind(this);
    }
    // adds music search results to favorites
    handleMusicAdd() {
        fetch('/postmusic', {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify({
                albumArt: this.state.results.artworkUrl100,
                artistName: this.state.results.artistName,
                songName: this.state.results.trackName,
                releaseDate: this.state.results.releaseDate
            })
        }).then(() => '', error => console.log(error));
        alert(`${this.state.results.trackName} added to Favorites`);


    }
    // adds music search results to favorites
    handleBooksAdd() {
        fetch('/postbooks', {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify({
                paperBack: this.state.results.artworkUrl100,
                author: this.state.results.artistName,
                title: this.state.results.trackName,
                genres: this.state.results.genres,
                releaseDate: this.state.results.releaseDate
            })
        }).then(() => '', error => console.log(error));
        alert(`${this.state.results.trackName} added to Favorites`);


    }
    // fetches data with specified media and search parameters when called
    fetchFile() {
        fetch(`/get/${this.state.search},${this.state.media}`)
            .then(apiRes => apiRes.json())
            .then(response => {
                console.log(response)
                this.setState({
                    results: response
                })
            })
    }

    // handles search request by user: fetches data related to info entered in search
    // bar.
    handleSearch(e) {
        e.preventDefault();
        // if search bar is empty when user clicks search
        // button, nothing should happen
        if (this.state.search === '') {
            return;
            // if search bar is populated, make fetch request
            // and set value for searchSentinel
        } else {
            this.setState({
                searchSentinel: '1'
            })
            this.fetchFile();
        }
    }
    // handles input of item of interest into search bar
    // and sets state item 'search' to it
    handleInput(e) {
        this.setState({
            search: e.target.value
        })
    }
    // handles selection of Music radio button
    handleMusic(e) {
        this.setState({
            media: e.target.value
        })
    }
    // handles selection of Books radio button
    handleBooks(e) {
        this.setState({
            media: e.target.value
        })
    }


    render() {
        return (
            <Container id='home-container'>
                <Card className="text-center">
                    <Card.Body>
                        <Card.Title id='title-home'><span id='istore'>iStore</span> <br /><span id='buscar'>Search</span> <br /> <span id='bruja'>Wizard</span></Card.Title>
                        <Card.Text>
                            <input id='search-bar' type='text' onChange={this.handleInput} value={this.state.search} placeholder='Search' />
                            <br />

                            <label id='books-radio'>
                                Books
                                <input className='radio-boton' onClick={this.handleBooks} type='radio' value='ebook' name='media' />
                            </label>
                            <label id='music-radio'>
                                Music
                                <input className='radio-boton' onClick={this.handleMusic} type='radio' value='music' name='media' />
                            </label>

                            <br />
                            <small>Please select either Music or Books before searching</small>
                        </Card.Text>
                        <Button id='home-search' onClick={this.handleSearch} variant="primary">Search</Button>
                    </Card.Body>
                </Card>
                {this.state.media === 'music' && this.state.results ?
                    <Music albumArt={this.state.results.artworkUrl100}
                        artistName={this.state.results.artistName}
                        songName={this.state.results.trackName}
                        releaseDate={this.state.results.releaseDate}
                        handleAdd={this.handleMusicAdd}
                    /> : ''}
                {this.state.media === 'ebook' && this.state.results ?
                    <Books paperBack={this.state.results.artworkUrl100}
                        author={this.state.results.artistName}
                        title={this.state.results.trackName}
                        genres={this.state.results.genres}
                        releaseDate={this.state.results.releaseDate}
                        handleAdd={this.handleBooksAdd}
                    /> : ''}
                {this.state.searchSentinel === '1' && this.state.results === undefined ? <h1>Item Not Found. Please Try Again</h1> : ''}
                {this.state.renderSentinel && <Favorites favList={this.state.favList} />}

            </Container>
        );
    }
}