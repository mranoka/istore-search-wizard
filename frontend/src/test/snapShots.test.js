import React from 'react';
import Home from '../Components/Home';
import Favorites from '../Components/Favorites';
import renderer from 'react-test-renderer';


test('Home renders correctly!', () => {
    const tree = renderer
        .create(<Home />).toJSON();
        expect(tree).toMatchSnapshot();
})
test('Favorites renders correctly!', () => {
    const tree = renderer
        .create(<Favorites />).toJSON();
        expect(tree).toMatchSnapshot();
})