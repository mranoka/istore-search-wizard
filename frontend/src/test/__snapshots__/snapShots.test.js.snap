// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites renders correctly! 1`] = `<div />`;

exports[`Home renders correctly! 1`] = `
<div
  className="container"
>
  <div
    className="text-center card"
  >
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        iStore 
        <br />
        Search 
        <br />
         Wizard
      </div>
      <p
        className="card-text"
      >
        <input
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
        <br />
        <label>
          Books
          <input
            name="media"
            onClick={[Function]}
            type="radio"
            value="ebook"
          />
        </label>
        <label>
          Music
          <input
            name="media"
            onClick={[Function]}
            type="radio"
            value="music"
          />
        </label>
        <br />
        <small>
          Please select either Music or Books before searching
        </small>
      </p>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
  
  
  
  
</div>
`;

exports[`renders correctly! 1`] = `
<div
  className="container"
>
  <div
    className="text-center card"
  >
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        iStore 
        <br />
        Search 
        <br />
         Wizard
      </div>
      <p
        className="card-text"
      >
        <input
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
        <br />
        <label>
          Books
          <input
            name="media"
            onClick={[Function]}
            type="radio"
            value="ebook"
          />
        </label>
        <label>
          Music
          <input
            name="media"
            onClick={[Function]}
            type="radio"
            value="music"
          />
        </label>
        <br />
        <small>
          Please select either Music or Books before searching
        </small>
      </p>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
  
  
  
  
</div>
`;

exports[`renders correctly! 2`] = `<div />`;
